src/xmpp/xmpp-im/xmpp_mamtask.cpp

```diff
@@ -0,0 +1,178 @@
+#include "xmpp_mamtask.h"
+
+using namespace XMLHelper;
+using namespace XMPP;
+
+class MAMTask::Private {
+public:
+    Jid       j;
+    QDateTime from;
+    QDateTime to;
+    bool      allowMUCArchives;
+    int mamPageSize; // TODO: this is the max page size for MAM request. Should be made into a config option
+    int       mamMaxMessages; // maximum mam pages total, also should be config
+    bool flipPages;
```

Comment:
nit: it's preferable to keep same data types close to each other. 
it will spare you a few bytes. 
In this case it's not really important because we don't call it often in parallel.